eclair-core/src/main/scala/fr/acinq/eclair/transactions/Transactions.scala

```diff
@@ -206,39 +207,97 @@ object Transactions {
 
   def decodeTxNumber(sequence: Long, locktime: Long): Long = ((sequence & 0xffffffL) << 24) + (locktime & 0xffffffL)
 
-  def makeCommitTx(commitTxInput: InputInfo, commitTxNumber: Long, localPaymentBasePoint: PublicKey, remotePaymentBasePoint: PublicKey, localIsFunder: Boolean, localDustLimit: Satoshi, localRevocationPubkey: PublicKey, toLocalDelay: CltvExpiryDelta, localDelayedPaymentPubkey: PublicKey, remotePaymentPubkey: PublicKey, localHtlcPubkey: PublicKey, remoteHtlcPubkey: PublicKey, spec: CommitmentSpec): CommitTx = {
+  /**
+   * Represent a link between a commitment spec item (to-local, to-remote, htlc) and the actual output in the commit tx
+
+   * @param output transaction output
+   * @param redeemScript redeem script that matches this output (most of them are p2wsh)
+   * @param specItem commitment spec item this output is built from
+   */
+  case class CommitmentOutputLink(output: TxOut, redeemScript: Seq[ScriptElt], specItem: CommitmentOutput)
+
+  object CommitmentOutputLink {
+    def sort(a: CommitmentOutputLink, b: CommitmentOutputLink): Boolean = (a.specItem, b.specItem) match {
+      case (DirectedHtlc(OUT, htlcA), DirectedHtlc(OUT, htlcB)) if htlcA.paymentHash == htlcB.paymentHash && htlcA.amountMsat == htlcB.amountMsat =>
```

Comment:
Why do we only do this for outgoing HTLCs and not incoming as well?

Reply:
The extra CLTV tie-breaking needs to be applied only to outgoing HTLC because they're the  ones that can be spent *by us* via an HTLC-timeout transaction. The HTLC-timeout transactions have the `cltvExpiry` in the locktime and different CLTV values will result in different transactions, this forces us to agree upon an ordering even when the offered HTLC outputs are identical. I've had some trouble putting this together myself, i think it deserves a comment in the code :)

Reply:
Oh yeah, gotcha, since in the claim-htlc-timeout for received HTLC the cltv is directly inside the script, BIP 69 is enough. This is subtle, thanks for the comment you added!